.github/ISSUE_TEMPLATE/new-release.md

```diff
@@ -0,0 +1,46 @@
+---
+name: New Release
+about: Propose a new release
+title: Release v0.x.0
+labels: ''
+assignees: ''
+
+---
+
+## Release Checklist
+- [ ] For major or minor releases (v$MAJ.$MIN.0), create a new release branch.
+  - [ ] A maintainer creates a vanilla release branch with
+        `git branch release-$MAJ.$MIN main`
+  - [ ] A maintainer pushes the new release branch with
+        `git push release-$MAJ.$MIN` (e.g., release-0.1)
+- [ ] Update things like README, deployment templates, docs, configuration, test/e2e flags.
+      Submit a PR against the release branch.
+- [ ] A maintainer [prepares a draft release](https://github.com/kubernetes-sigs/gateway-api-inference-extension/releases)
+  - [ ] Write the change log into the draft release.
```

Comment:
Will this project follow the GW API approach to changelog management ([xref](https://github.com/kubernetes-sigs/gateway-api/tree/main/CHANGELOG))?

Reply:
Can you please suggest what the exact step that we should add here?

Reply:
I'll defer to @robscott since he understands how GW API performs changelog management.

Reply:
I'd recommend incorporating something like this: https://github.com/kubernetes-sigs/gateway-api/blob/main/RELEASE.md#writing-a-changelog